### PR TITLE
Remove need for `mergeState`

### PR DESCRIPTION
**Pull request by axelboc**
As per their name, `filterables` represent things that can be filtered. They are _static_ definition objects, so by definition I don't see them as a "state" but as a configuration. This is the premise for this PR.

Currently `mergeState` takes two arrays: the `fitlerables`  and the `filters` and returns a single array that has the same length as `filters` where the objects contain the filter values from `filters` (`value`) and the other static properties from `filterables` (`operator`, `name`, etc.)

This PR removes the `mergeState` function completely and replaces it with a single line that maps through the `filters` and, for each filter object, merges it with its corresponding filterable definition object. I managed to get this opertion down to one line by simply storing the filterables by label - i.e. `{ 'do-type': { label: 'do-type', name: 'type', ... }`.

### CODE REVIEW

File: src/App/adapter/init.js [hunk 1]
```diff
@@ -19,11 +19,22 @@ const makeLabelCandidates = (obj) => {
...
+function init(groupedFilterables) {
+  const filterablesArr = Object.entries(groupedFilterables)
+    .flatMap(([group, arr]) => arr.map((f) => ({ ...f, group }))) // flatten filterables, while keeping a reference to their groups
+    .map(makeLabelCandidates) // generate missing labels
+    .reduce(uniqueLabelReducer, []) // make sure labels are unique
+    // add specific operators
+    .map((obj) =>
+      obj.range
+        ? { ...obj, operator: 'between' }
+        : ['title', 'sample_chemical_formula'].includes(obj.name)
```
Comment (axelboc) [inline src/App/adapter/init.js:31]: I've moved this here, as it has to be done before the line below. Obviously, if we want the adapter to be generic, it should not have to know about `title` and `sample_chemical_formula`. I'm hoping we can add a property to the relevant `filterables` that we can use to identify this use case in the same way as `obj.range` above. I will work on this next.

File: src/App/adapter/lib/state.js [hunk 2]
```diff
@@ -63,27 +62,3 @@ export function parseState(state, config) {
...
-
-export function mergeState(inits, diffs) {
-  const useful = diffs.reduce(
-    (acc, scope) => [...acc, scope.label, scope.group],
-    [LABEL_FOR_CONFIG],
-  )
-  const usefulInits = inits.filter(({ label }) => useful.includes(label))
-  const byLabel = groupByLabel([...usefulInits, ...diffs])
-  return Object.entries(byLabel).flatMap(([, a]) => squash(a))
-}
```
Comment (axelboc) [inline src/App/adapter/lib/state.js:89]: If this code was dealing with an edge case of some sort that I have missed, please do let me know and I'll add a test case for it.

File: src/App/adapter/translate.js [hunk 3]
```diff
@@ -1,19 +1,21 @@
...
   order: undefined, // e.g.`['foo ASC', 'bar DESC']`
 }
 
-function translate(diffState, initialState = [], queryConfig = {}) {
+function translate(filters, queryConfig = {}) {
   const config = { ...DEFAULT_CONFIG, ...queryConfig }
+  const { filterables, order } = config
 
-  const state = mergeState(initialState, diffState)
+  const state = filters.map((f) => ({ ...filterables[f.label], ...f }))
```
Comment (axelboc) [inline src/App/adapter/translate.js:18]: One-liner thanks to `filterables` being now stored by label.

File: src/Document/DocumentPage.js [hunk 4]
```diff
@@ -13,7 +13,6 @@ function DocumentPage() {
       ['datasets', 'instrument'],
       ['members', 'person'],
     ],
-    limit: false,
```
Comment (axelboc) [inline src/Document/DocumentPage.js:16]: I believe this is the default behaviour.

File: src/Search/Search.js [hunk 5]
```diff
@@ -3,23 +3,24 @@ import shallow from 'zustand/shallow'
 
 import { useAppStore } from '../App/stores'
 import { Card, Flex, Text, Button } from '../Primitives'
-import { initialFilters } from '../filters'
+import { filterables } from '../filters'
 import FilterGroup from './FilterGroup'
 
 const ORDER = ['title', 'type', 'keywords']
 const sortFilters = (a, b) => ORDER.indexOf(a?.name) - ORDER.indexOf(b?.name)
+const filterablesArr = Object.values(filterables)
```
Comment (axelboc) [inline src/Search/Search.js:11]: Since `fitlerables` are now stored by label, I just need to put them in an array here. I'm not too worried about this code, as it will likely change once we start worrying about making the UI more configurable.
